### ISSUE TITLE
Create a robots.txt file

### ISSUE THREAD

**Issue by Hypolite** [friendica/friendica#3334]
In my web server log I'm getting quite a few 404 errors about a missing robots.txt.

Having a blank robots.txt would be great to prevent such errors to happen.

Incidentally, are there any paths that robots shouldn't probe?

Useful reference: http://www.robotstxt.org/robotstxt.html

**Comment by annando** [comment]
I'm having this robots.txt on my friendica installation:
```
User-agent: Twitterbot
Disallow:

User-agent: ltx71
Disallow: /

User-agent: *
Disallow: /

user-agent: AhrefsBot
disallow: /
Crawl-Delay: 60
```

I don't want any crawlers on my system at all. But the problem is that there maybe admins who don't think so. Having a central robots.txt would force our users to obey to a central decision about this.

**Comment by Quix0r** [comment]
You should handle then the request `/robots.txt` in `index.php` so every admin can easily setup an own robots.txt.

**Comment by Hypolite** [comment]
Yep, we can also have an admin page where the robots.txt content can be edited, the default being empty.

The idea would be to commit an empty file and then ignore further changes in git, I hope it's feasible.

**Comment by Quix0r** [comment]
Easier:
- add robots.txt to .gitignore and don't commit it
- in admin area check if robots.txt is found
- if not, show message that you may want to create it
- provide common URLs you usually don't want get crawled (like country flags, smileys, imprint, et cetera)

**Comment by Shnoulle** [comment]
Didn't need any hack htaccess do the job

`RewriteCond %{REQUEST_FILENAME} !-f`

If there are a robot.txt file : get the existing, if not go to index.

I take this one to start (staring by undersanding the path/url system ... not the best ;) )

**Comment by Hypolite** [comment]
You can model the `robots.txt` controller from `mod/friendica.php`.

**Comment by fabrixxm** [comment]
you can create `mod/robots_txt.php` and it will called on `http://domain/robots.txt` if a `robots.txt` file doesn't exists (as per `.htaccess` config)

**Comment by Hypolite** [comment]
Closed via #3393, thanks to @Shnoulle !
